clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -559,6 +559,9 @@ private void handleFetchResponse(ClientResponse resp, FetchRequest request) {
                 MemoryRecords records = MemoryRecords.readableRecords(buffer);
                 List<ConsumerRecord<K, V>> parsed = new ArrayList<>();
                 for (LogEntry logEntry : records) {
+                    // Skip the messages earlier than current position.
+                    if (logEntry.offset() < position)
+                        continue;
```

Comment:
Nitpick: it would be more composable to do:

```java
if (logEntry.offset() >= position) {
...
}
```

The reason is that we can freely move the if block to another method (if it needs to be reused or becomes more complex) and the code still works correctly. `continue` and `break` couple code to the surrounding loop (unnecessarily in many cases).